pywinauto/controls/uia_controls.py

```diff
@@ -219,3 +228,199 @@ def item_count(self):
+            if pos_end is None and not isinstance(start, six.string_types):
+                pos_end = end
+        else:
+            pos_start = 0
+            pos_end = len(self.window_text())
+
+        if isinstance(text, six.text_type):
+            if six.PY3:
+                aligned_text = text
+            else:
+                aligned_text = text.encode(locale.getpreferredencoding())
+        elif isinstance(text, six.binary_type):
+            if six.PY3:
+                aligned_text = text.decode(locale.getpreferredencoding())
+            else:
+                aligned_text = text
+        else:
+            # convert a non-string input
+            if six.PY3:
+                aligned_text = six.text_type(text)
+            else:
+                aligned_text = six.binary_type(text)
+
+        # Set text using IUIAutomationValuePattern
+        iface = uia_defs.get_elem_interface(self.element_info.element, "Value")
+        # Calculate new text value
+        current_text = self.window_text()
+        new_text = current_text[:pos_start] + aligned_text + current_text[pos_end:]
+        iface.SetValue(new_text)
+
+        #win32functions.WaitGuiThreadIdle(self)
+        #time.sleep(Timings.after_editsetedittext_wait)
+
+        if isinstance(aligned_text, six.text_type):
+            self.actions.log('Set text to the edit box: ' + aligned_text)
+        else:
+            self.actions.log(b'Set text to the edit box: ' + aligned_text)
+
+        # return this control so that actions can be chained.
+        return self
+    # set SetText as an alias to set_edit_text
+    set_text = set_edit_text
+
+    #-----------------------------------------------------------
+    def select(self, start = 0, end = None):
+        """Set the edit selection of the edit control"""
+        self.verify_actionable()
+        self.set_focus()
+
+        # if we have been asked to select a string
+        if isinstance(start, six.text_type):
+            string_to_select = start
+        elif isinstance(start, six.binary_type):
+            string_to_select = start.decode(locale.getpreferredencoding())
+        elif isinstance(start, six.integer_types):
+            if isinstance(end, six.integer_types) and start > end:
+                start, end = end, start
+            string_to_select = self.window_text()[start:end]
+
+        if string_to_select:
+            elem = self.element_info.element
+            iface = uia_defs.get_elem_interface(elem, "Text")
+            document_range = iface.DocumentRange
+            search_range = document_range.FindText(string_to_select, False, False)
+
+            try:
+                search_range.Select()
+            except ValueError:
+                raise RuntimeError("Text '{0}' hasn't been found".format(string_to_select))
+
+            # give the control a chance to catch up before continuing
+            if self.element_info.handle:
+                win32functions.WaitGuiThreadIdle(self)
```

Comment:
I'm not sure that waiting for something is really necessary here. Probably `Select()` method of the pattern should be a blocking call. So returning from it means all actions are done, isn't it?